Upgrade Maven Wrapper to version 3.3.2

aowss opened the pull request:
### Why <!-- Briefly describe why these changes are needed -->

Keep up to date with Maven upgrades

### What <!-- Briefly describe how these changes solve the problem -->

Just ran `mvn wrapper:wrapper`

orende commented on .github/workflows/pipeline.yml:
```diff
@@ -13,11 +13,11 @@ jobs:
+        uses: actions/setup-java@v4
         with:
-          java-version: '17'
+          java-version: '23'
```
We've chosen to stick with Java 17 until its end of premier support date (2026-09-30) in order to allow the largest possible range of up-to-date JDK users to use this project's code unmodified.

aowss replied:
that makes sense

aowss replied:
i also changed it back in the pom in that case

orende commented on src/main/java/se/citerus/dddsample/interfaces/tracking/ws/CargoTrackingDTOConverter.java:
```diff
@@ -16,7 +16,7 @@
 public class CargoTrackingDTOConverter {
     private static final DateTimeFormatter formatter = DateTimeFormatter
-            .ofLocalizedDateTime(FormatStyle.MEDIUM)
+            .ofPattern("MMM d, uuuu, h:mm:ss a", Locale.ENGLISH) // See https://github.com/spring-projects/spring-framework/wiki/Date-and-Time-Formatting-with-JDK-20-and-higher#recommendations
```
This is a good catch (FormatStyle.MEDIUM does seem like a bad choice given it's locale-dependent behavior) but how about using an ISO-8601 date pattern like `YYYY-MM-DDTHH:MM:SSZ` instead?

aowss replied:
imho, the ISO format is way better but i thought you had a specific reason for choosing this format.

aowss replied:
I used the ISO format for values but kept this format for descriptions since it's more readable

orende commented on pom.xml:
```diff
@@ -73,16 +73,17 @@
-        <developerConnection>scm:git:https://github.com/citerus/dddsample-core</developerConnection>
-        <url>https://github.com/citerus/dddsample-core</url>
+        <developerConnection>scm:git:https://github.com/aowss/dddsample-core</developerConnection>
+        <url>https://github.com/aowss/dddsample-core</url>
```
This line should stay as `https://github.com/citerus/dddsample-core`.

aowss replied:
of course

orende commented on pom.xml:
```diff
@@ -73,16 +73,17 @@
     <scm>
-        <developerConnection>scm:git:https://github.com/citerus/dddsample-core</developerConnection>
-        <url>https://github.com/citerus/dddsample-core</url>
+        <developerConnection>scm:git:https://github.com/aowss/dddsample-core</developerConnection>
```
This line should stay as `https://github.com/citerus/dddsample-core`.

aowss replied:
of course

orende reviewed (changes requested):
Thank you for the contribution! There's a couple of comments to be addressed below, but most of these changes look great.

aowss commented on pom.xml:
```diff
@@ -8,21 +8,21 @@
         Domain Driven Design implemented in Enterprise Java.
     </description>
-    <url>http://dddsample.sourceforge.net</url>
+    <url>https://github.com/aowss/dddsample-core</url>
```
I am sorry, this was due to the fact that i decided to do some more changes in my fork and accidentally included that.

aowss commented on pom.xml:
```diff
@@ -8,21 +8,21 @@
-            <url>scp://shell.sf.net/home/groups/d/dd/dddsample/htdocs/preview</url>
-        </site>
-    </distributionManagement>
+<!--    <distributionManagement>-->
```
ok

aowss commented:
@orende I think i have addressed all your comments
I imagine you noticed but I removed Gradle. I wanted to draw your attention to that to make sure you are OK with it.

orende commented:
I noticed you removed Gradle, and I have no objections to it since we haven't been able to effectively keep the gradle files up-to-date.
